app/interactors/delete_user.rb

```diff
@@ -0,0 +1,19 @@
+class DeleteUser
+  include Interactor
+
+  def call
+    validate_input
+    execute
+  end
+
+  private
+
+  def validate_input
+    context.fail!(error: "invalid input") unless context.id
+  end
+
+  def execute
+    context.user = User.find(context.id)
+    context.fail!(error: "user not deleted") unless context.user.destroy
```

Comment:
Remember that `execute`, or for that matter `call`, is not the final return value. The Interactor gem always returns the `context` so it doesn't matter what the last line of your `call/execute` method is or even if you do an explicit `return`.

I like using the same structure on all interactors. It's all about staying consistent for me :stuck_out_tongue:. It makes things easier to read. Just like your standard Rails conventions.